delivery_carrier_label_default/tests/test_print_label.py

```diff
@@ -1,28 +1,50 @@
-# -*- coding: utf-8 -*-
-# Copyright 2016 Camptocamp SA
+# Copyright 2013-2018 Camptocamp SA
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+import mock
+import base64
+
 from openerp.tests import common
 
+from .common import HTMLRenderMixin
+
+MOD_PATH = 'odoo.addons.base_delivery_carrier_label.models'
+LABEL_MODEL = MOD_PATH + '.shipping_label.ShippingLabel'
+
+
+class TestPrintLabel(common.TransactionCase, HTMLRenderMixin):
+    """When running tests Odoo 11 renders PDF reports as HTML, so we are
+    going to test the shipping labels as HTML document only.
+    """
 
-class TestPrintLabel(common.TransactionCase):
+    def check_label_content(self, b64_datas):
+        html_datas = base64.b64decode(b64_datas)
+        node = self.to_xml_node(html_datas)[0]
+        for div_class in ['page', 'address', 'recipient']:
+            tags = self.find_div_class(node, div_class)
+            self.assertEqual(len(tags), 1)
 
-    def test_print_default_label(self):
+    @mock.patch(LABEL_MODEL + '._selection_file_type')
```

Comment:
as the mocking is required for all tests you can move this to `setUp` + add a comment saying why you patch it.

Reply:
it doesn't work, I suppose that once we left the `setUp` method, the mocking disappears and is not available on `test_` methods. Looking for another solution.

Reply:
I implemented a decorator to patch the field, a half way solution.